File: src/main/java/com/matamercer/microblog/models/entities/UserKeyPair.java
package com.matamercer.microblog.models.entities;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "UserKeyPairs")
@Getter
@Setter
public class UserKeyPair extends BaseModel{
    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(nullable = false, length = 2048)
    private String publicKey;

    @Column(nullable = false, length = 2048)
    private String privateKey;

    public UserKeyPair(){

    }

    public UserKeyPair(User user, String publicKey, String privateKey) {
        this.user = user;
        this.publicKey = publicKey;
        this.privateKey = privateKey;
    }
}

File: src/main/java/com/matamercer/microblog/models/entities/UserKeyPair.kt
package com.matamercer.microblog.models.entities

import javax.persistence.*

@Entity
@Table(name = "user_key_pairs")
class UserKeyPair(
    @JoinColumn(name = "userKeyPairs")
    @ManyToOne
    private val user: User? = null,

    @Column(
        nullable = false,
        length = 2048
    )
    val publicKey: String? = null,

    @Column(nullable = false, length = 2048)
    private val privateKey: String? = null
) : BaseModel()